client/src/services/VideoService.js

```diff
@@ -0,0 +1,70 @@
+import AWS from 'aws-sdk';
+
+const S3_PUBLIC_KEY = import.meta.env.VITE_AWS_S3_PUBLIC_KEY ?? '';
+const S3_SECRET_KEY = import.meta.env.VITE_AWS_S3_SECRET_KEY ?? '';
```

Comment:
:memo: Need a better way to import these variables. This was temporary for my use - not pushing my AWS credentials for obvious reasons so this will change in future

Reply:
Yea good point.

Few ideas

1) Is it possible to generate some sort of one time token using the AWS API that grants user access?

2) Maybe we could make a request to the server that sends back the secret key, or both before the videoUpload is called?

Note on that suggestion I'm a little confused when the keys are used. Is a connection to the bucket remade each request? I guess when does AWS.config.update run?

I guess my idea is the environment variables are stored on our AWS instance, and retrieved by our server. When a client wants to upload a video first the keys are requested from our server and then used one time by the client to upload the video to AWS.

Reply:
Er sorry I guess I should've been more clear on this comment. So I've created a AWS Role which acts as an account. The role has S3 abilities. The access keys for this role are something I generated (similar to how you'd generate your own ssh keys or something of the like) and are static.

The `VideoService` is instantiated during the build with those static values i.e. they won't change. I need to look to see if there's a clean way to inject the env variables into our dev environments remotely rather than giving everyone a file with the credentials on it to have locally (just as a safeguard in case someone accidentally commits the creds). I can inject the env variables from Github on production builds no issue.

Your way of getting a one time access key would be the super proper way but that requires bunch of a setup and I wasn't positive it would still be within AWS free tier so I scrapped the idea lol. But yeah that would be the safer way if this was more official

Reply:
I see what your saying. So it's an issue of each of us having the key for local development. If those keys are instantiated on build is there still some kind of script we could write that fetches those keys from a secure AWS service?

I asked the homie chatGPT about our situation and this is what it came up with

Q; "On our project, we have one aws account set up and private and secret keys are utilized in one place on our fronted. We want to avoid all having a file with the secret information for local development and want to know if there is a way to remotely have access to these environment variables without them being on our system. When its built for production that env vars can be injected through github. But we are looking for a solution for local development. 

some more context,

---

"The VideoService is instantiated during the build with those static values i.e. they won't change. I need to look to see if there's a clean way to inject the env variables into our dev environments remotely rather than giving everyone a file with the credentials on it to have locally (just as a safeguard in case someone accidentally commits the creds). I can inject the env variables from Github on production builds no issue"

What are some possible solutions? are frontend is using react"

A:
Environment Variable Injection via Backend: You can set up your backend to fetch the required environment variables from a secure location, such as AWS Parameter Store or Secrets Manager. During local development, you can configure your local backend to fetch these variables from the same location. Your frontend can then make requests to your backend to retrieve these variables when needed.

Proxy Server: Set up a proxy server that sits between your frontend and AWS services. This proxy server can be configured to inject the necessary environment variables before forwarding requests to AWS services. During local development, you can run a local instance of the proxy server and configure it to fetch the environment variables from a secure location.

Environment Variable Injection via Build Scripts: Create build scripts that inject the required environment variables during the build process. You can use tools like dotenv to manage environment variables during development. These scripts can fetch the variables from a secure location and inject them into your application during the build phase.

Use AWS Systems Manager Parameter Store: AWS Systems Manager Parameter Store provides a centralized store to manage your configuration data, whether plain-text data such as database strings or secrets such as passwords. You can use this service to store your environment variables securely and fetch them in your application during runtime.
---